presto-bigquery/src/main/java/com/facebook/presto/plugin/bigquery/BigQueryEmptyProjectionPageSource.java

```diff
@@ -0,0 +1,85 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.plugin.bigquery;
+
+import com.facebook.airlift.log.Logger;
+import com.facebook.presto.common.Page;
+import com.facebook.presto.common.PageBuilder;
+import com.facebook.presto.spi.ConnectorPageSource;
+import com.google.common.collect.ImmutableList;
+
+import java.io.IOException;
+
+public class BigQueryEmptyProjectionPageSource
+        implements ConnectorPageSource
+{
+    private static final Logger log = Logger.get(BigQueryEmptyProjectionPageSource.class);
+    private final long numberOfRows;
+    private boolean finished;
+
+    public BigQueryEmptyProjectionPageSource(long numberOfRows)
+    {
+        this.numberOfRows = numberOfRows;
+        this.finished = false;
+    }
+
+    @Override
+    public long getCompletedBytes()
+    {
+        return 0;
+    }
+
+    @Override
+    public long getCompletedPositions()
+    {
+        return finished ? numberOfRows : 0;
+    }
+
+    @Override
+    public long getReadTimeNanos()
+    {
+        return 0;
+    }
+
+    @Override
+    public boolean isFinished()
+    {
+        return finished;
+    }
+
+    @Override
+    public Page getNextPage()
+    {
+        log.warn("[%s] creating %d empty rows", Thread.currentThread(), numberOfRows);
```

Comment:
do we need this log?

Reply:
this might provide a better understanding how much data is coming from each split, downgrade to debug if ok with you?